packages/remix-dev/cli/run.ts

```diff
@@ -303,6 +316,22 @@ export async function run(argv: string[] = process.argv.slice(2)) {
     case "setup":
       await commands.setup(input[1]);
       break;
+    case "codemod": {
+      if (!flags.dry) {
+        checkGitStatus(flags.force);
+      }
+
+      let answers = await commands.codemod.questions({
+        input: {
+          files: input[2],
+          transform: input[1],
+        },
+        showHelp,
+      });
+
```

Comment:
If we ask for a `projectDir` (like I mentioned in my other comment), we can use that here to find the `package.json` file and do all the updates to it that we need. We can also auto-detect the server runtime and server adapters from the dependencies listed in `package.json`.